.circleci/config.yml

```diff
@@ -22,13 +24,24 @@ jobs:
     <<: *defaults
     docker:
       - image: golangci/golangci-lint:v1.43-alpine
+    resource_class: medium+
     steps:
       - checkout
+      - *restore_mod
+      - restore_cache:
+          keys:
+            - v1-buildcache-
+          paths:
+            - /root/.cache
+      # We want to keep the cache, but save it every time
+      - run: echo "$CIRCLE_BUILD_NUM-$CIRCLE_JOB" > /tmp/build_num
       - run: golangci-lint run
-      # Since golangci is ignoring test files, make sure gofmt gets run on
-      # everything
-      - run: golangci-lint run --tests=true --disable-all --enable=gofmt --enable=gofumpt --no-config
+      - save_cache:
+          key: v1-buildcache-{{ checksum "/tmp/build_num" }}
```

Comment:
this will change every time, but still drastically speeds things up when no dependencies have changed.